File: Javascript Basics/fibonacci.js
//Fibonacci sequence

function fibonaaci(x)
  {
    var ans=[];
    
    if(x===1)
    {ans.push(0);}
    else if (x==2)
    {ans.push(0,1);}
    else //if(x > 2)
    {ans=[0,1];
      for(var i=2;i<x;i++)
      { var n=ans.length;
      var next = ans[n-2] + ans[n-1];
      ans.push(next);
      }
    }
    return ans;
  }
fibonaaci(8);

File: RnBT_course/Lecture 1/fibonacci.cpp
// fibonacci series : 0,1,1,2,3,5,8,13...
//TC : O(2^n)
//SC : O(n)
#include <iostream>
#include <bits/stdc++.h>
using namespace std;
int fibonacci(int n)
{ // base case
    if (n == 0)
    {
        return 0;
    }
    if (n == 1 || n == 2)
    {
        return 1;
    }
    //recursive relation
    int ans = fibonacci(n - 1) + fibonacci(n - 2);
    return ans;
}
int main()
{
    int n = 3;
    cout << n << "th term : " << fibonacci(n) << endl;
    return 0;
}